### ISSUE TITLE
Pattern mask with number in start ignores digits after first space

### ISSUE THREAD

**Issue by aktanoff** [uNmAnNeR/imaskjs#888]
**Describe the bug**
When using pattern mask with number + currency symbol after number - when pasting value, or set defaultValue imask ignores number parts after first thousands separator symbol

Problematic mask example:
```
{
  blocks: {
    d: {
      mask: Number,
      scale: 2,
      signed: false,
      thousandsSeparator: " ",
      padFractionalZeros: false,
      normalizeZeros: false,
      radix: ",",
      mapToRadix: [".", "б", "^", "ю", "&"]
    }
  },
  mask: `d $`,
  lazy: false
}
```

**To Reproduce**
Example:
https://codesandbox.io/s/jolly-williams-86tkvv?file=/src/App.tsx

**Expected behavior**
IMask shouldn't ignore digits after first space

**Environment:**
 - OS: MacOS 13.4 (22F66)
 - Browser chrome, safari, firefox
 - Version any
 - IMask from 6.6.1 up to 7.0.1
 - Framework/plugin experiencing it in pure imask and in react-imask

**Additional context**
Space in pattern mask don't affect the problem

Can you please also add support for non breakable space as thousandsSeparator?
https://unicode-explorer.com/c/00A0

**Comment by aktanoff** [comment]
And also found bug with defaultValue initialization with mask like this
```
const PRICE_INPUT_MASK_SETTINGS = {
  mask: Number,
  scale: 2,
  signed: false,
  thousandsSeparator: " ",
  padFractionalZeros: true,
  normalizeZeros: true,
  radix: ",",
  mapToRadix: [".", "б", "^", "ю", "&"]
};
```

https://codesandbox.io/s/zealous-dan-39wwpc?file=/src/App.tsx

**Comment by mlshvdv** [comment]
I met the same bug

**Comment by uNmAnNeR** [comment]
@aktanoff 
> Can you please also add support for non breakable space as thousandsSeparator?
https://unicode-explorer.com/c/00A0

seems like `thousandsSeparator: " ",` works fine, does not it? Here is actually "U+00A0" inserted from the link you provided. Produces:
![image](https://github.com/uNmAnNeR/imaskjs/assets/1927921/cf7b3f21-6d7a-467a-8e05-c6a37bb85adb)

**Comment by aktanoff** [comment]
> And also found bug with defaultValue initialization with mask like this
> 
> ```
> const PRICE_INPUT_MASK_SETTINGS = {
>   mask: Number,
>   scale: 2,
>   signed: false,
>   thousandsSeparator: " ",
>   padFractionalZeros: true,
>   normalizeZeros: true,
>   radix: ",",
>   mapToRadix: [".", "б", "^", "ю", "&"]
> };
> ```
> 
> https://codesandbox.io/s/zealous-dan-39wwpc?file=/src/App.tsx

@uNmAnNeR, hi, didn't you fix this bug?
should i create another issue for that?
here i'm just using unmaskedValue as defaultValue, looks like it should be working fine
